### ISSUE TITLE
[a11y]: `SidePanel` stories should use `h3` for subheadings since side panel title is `h2`

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#5767]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

`2.45.0-rc.1`

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS

### Description

The example subheaders in the body of the `SidePanel` should be changed from `h5` to `h3` so that there is a logical and expected order of headings within the panel.

### WCAG 2.1 Violation

1.3.1 Info and Relationships

### Reproduction/example

Visible across all side panel stories

### Steps to reproduce

[Visit side panel stories](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
